src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -400,10 +400,10 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelinesRightLe
 
     left->pipe.collected_processors = collected_processors;
 
-    /// Collect the NEW processors for the right pipeline.
-    QueryPipelineProcessorsCollector collector(*right);
     /// Remember the last step of the right pipeline.
     IQueryPlanStep * step = right->pipe.processors->back()->getQueryPlanStep();
+    /// Collect the NEW processors for the right pipeline.
```

Comment:
why it was needed?

Reply:
if we not add step in collector, those processors will not set stepName

Reply:
yep, sorry, I missed that we now pass `step` as a parameter